wayland-display-core/src/wayland/handlers/wl_drm.rs

```diff
@@ -1,3 +1,24 @@
-use crate::{comp::State, wayland::protocols::wl_drm::delegate_wl_drm};
+use crate::{comp::State, wayland::protocols::wl_drm::{delegate_wl_drm, DrmHandler, ImportError}};
+use smithay::{
+    backend::{allocator::dmabuf::Dmabuf, drm::DrmNode},
+    reexports::wayland_server::{protocol::wl_buffer::WlBuffer, Resource},
+    wayland::dmabuf::DmabufGlobal,
+};
+use smithay::backend::renderer::ImportDma;
+
+impl<D: 'static> DrmHandler<Option<D>> for State {
+    fn dmabuf_imported(
+        &mut self,
+        global: &DmabufGlobal,
+        dmabuf: Dmabuf,
+    ) -> Result<Option<D>, ImportError> {
+        todo!("self.renderer.import_dmabuf(&dmabuf, None) ???");
+        Err(ImportError::Failed)
+    }
+
+    fn buffer_created(&mut self, buffer: WlBuffer, result: Option<D>) {
+        todo!()
+    }
```

Comment:
I've added the trait `DrmHandler` but I'm a bit lost on how to actually implement this. I was trying to 
```rust
self.renderer.import_dmabuf(&dmabuf, None)
```
but I'm not sure what should I return here.. I've tried `Option<DrmNode>` at first but then it was complaining that
> the trait bound `comp::State: DrmHandler<comp::State>` is not satisfied

I like this approach because everything else in the wl_drm implementation stay the same..

Reply:
Oh right, the returned value isn't relevant for this compositor (we don't deal with multiple gpus). So just replace the `Option<D>` with `()` (that is the unit type, aka an empty tuple. Used in place of "Nothing").

`buffer_created` can be an empty implementation and `self.renderer.import_dmabuf(&dmabuf, None).map(|_| ()).map_err(|_| ImportError::Failed)` should work for the `dmabuf_imported` method.